os/board/rtl8730e/src/rtl8730e_mipi_lcdc.c

```diff
@@ -269,15 +272,66 @@ void rtl8730_lcdc_initialize(void)
 	config.mipi_dsi_VSA = MIPI_DSI_VSA;
 	config.mipi_lcd_limit = MIPI_LCD_LIMIT;
 	config.lcd_lane_num = MIPI_LANE_NUMBER;
-	rtl8730_gpio_reset();
+	rtl8730e_gpio_reset();
 	struct mipi_dsi_host *dsi_host = (struct mipi_dsi_host *)amebasmart_mipi_dsi_host_initialize(&config);
 	struct mipi_dsi_device *dsi_device = (struct mipi_dsi_device *)mipi_dsi_device_register(dsi_host, "dsi", 0);
-	struct lcd_dev_s *dev = (struct lcd_dev_s *)mipi_lcdinitialize(dsi_device, &g_rtl8730_config_dev_s.lcd_config);
+	struct lcd_dev_s *dev = (struct lcd_dev_s *)mipi_lcdinitialize(dsi_device, &g_rtl8730e_config_dev_s.lcd_config);
 	LcdcInitValues(config);
-	rtl8730_lcd_init();
+	rtl8730e_lcd_init();
+	rtl8730e_enable_lcdc();
+	rtl8730e_control_backlight(CONFIG_LCD_MAXPOWER);
+
 	if (lcddev_register(dev) < 0) {
 		lcddbg("ERROR: LCD driver register fail\n");
 		return;
 	}
 	lcdvdbg("LCD driver register success\n");
 }
+
+#ifdef CONFIG_PM
+void rtl8730e_lcdc_pm(enum pm_state_e pmstate)
+{
+	/* Get current PM state first */
+	enum pm_state_e curr_state = pm_querystate(PM_IDLE_DOMAIN);
+	/* For LCDC with PM */
+	int fd;
+	int ret;
+	fd = open("/dev/lcd0", O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		lcddbg("ERROR: Failed to open lcd port : error:%d\n", fd);
+		return;
+	}
+	switch (pmstate) {
+		case PM_NORMAL:
+			/* Need to check which state are we transitioning from */
+			/* PM_IDLE: Set the backlight to max power */
+			/* PM_STANDBY: Run LCD */
+			/* PM_SLEEP: Nothing to do here, MIPI/LCDC will be reinit during wakeup */
+			if (curr_state == PM_STANDBY) {
+				rtl8730e_enable_lcdc();
+			}
+			rtl8730e_control_backlight(CONFIG_LCD_MAXPOWER);
+			break;
+		/* Backlight dimming */
+		case PM_IDLE:
+			rtl8730e_control_backlight(50);
```

Comment:
I think it would be good to have a dimming control for LCDs in PM regardless of the type of LCD.

To achieve this, I believe we should utilize the HAL function of the LCD driver to control the LCD dimming.

Please share your thoughts on this matter.
is there better functionality for power management in terms of LCDC?

Reply:
Hi @ewoodev , after the PM structure change (ie. removed pm_register), BSP driver would not know the PM state info anymore. In return, this implementation need to be reconsidered. Let's discuss about PM + LCD again after we completed other pending PRs.